chain/store/store.go

```diff
@@ -730,38 +737,82 @@ func (cs *ChainStore) MessagesForTipset(ts *types.TipSet) ([]types.ChainMsg, err
 		return nil
 	}
 
-	var out []types.ChainMsg
+	selectMsg := func(m *types.Message) (bool, error) {
+		if err := preloadAddr(m.From); err != nil {
+			return false, err
+		}
+
+		if applied[m.From] != m.Nonce {
+			return false, nil
+		}
+		applied[m.From]++
+
+		// TODO: We should be able to take this out -- such a message can never enter mempool or a valid block
+		if balances[m.From].LessThan(m.RequiredFunds()) {
```

Comment:
Such message can be in block and for sure can be in mpool.
It should also be executed so miner pays the penalty.

Reply:
yeah, you're right...fixed